src/main/kotlin/com/openlattice/data/storage/PostgresQueries.kt

```diff
@@ -437,10 +420,31 @@ internal fun buildFilterClause(fqn: String, filter: Set<Filter>): String {
 internal fun selectEntityKeyIdsByLinkingIds( linkingIds:Set<UUID> ): String {
     val linkingEntitiesClause = buildLinkingEntitiesClause(linkingIds)
     return "SELECT ${LINKING_ID.name}, array_agg(${ID_VALUE.name}) AS ${ENTITY_KEY_IDS.name} " +
-            "FROM ${selectEntityKeyIdsWithCurrentVersionSubquerySql(linkingEntitiesClause, setOf(), true)} " +
+            "FROM  ${IDS.name} " +
+            "WHERE ${VERSION.name} > 0 AND ${LINKING_ID.name} IS NOT NULL $linkingEntitiesClause " +
             "GROUP BY ${LINKING_ID.name} "
 }
 
 internal fun buildLinkingEntitiesClause(linkingIds:Set<UUID>): String {
-    return " AND ${LINKING_ID.name} IN ( ${linkingIds.joinToString(",") { "'$it'" }} )"
+    return " AND ${LINKING_ID.name} IN ( ${linkingIds.joinToString(",") { "'$it'" }} ) "
+}
+
+private fun getJoinColumns(linking: Boolean, omitEntitySetId: Boolean): List<String> {
+    return if (linking) {
+        if(omitEntitySetId) {
+            listOf(LINKING_ID.name)
```

Comment:
When do we omit entity set id?

Reply:
When we want the data merged by linking id: searching, getting a linking entity, getting the whole linking entity set.
This way we don't have to afterprocess and merge all properties by linking id

Reply:
Let me clarify, why do we omit instead of doing an array aggregate?

Reply:
You mean add an extra outer select?

Reply:
If we're doing a group by linking id, we can also collect array_agg(entity_set_id) in the same clause. I think this is fine for now-- we may have to modify when we need to return the data source along with the entity.